### PR TITLE
Eslint improvements.

### PR DESCRIPTION
**Pull request by aasimkhan30**
1. Running eslint and localization in parallel during pre-commit hook for better perf.
2. Upgrading eslint to latest
3. Adding caching to eslint to make subsequent runs faster.  This also involved removing lint gulp task and using cli to get caching option.
4. Adding eslint cache file to ignore.
5. Re-enable prettier as error. It was set to warning leading to some files not being formatted correctly.
6. Reformatting those files.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,12 +53,13 @@
     "build": "gulp build",
     "compile": "gulp ext:compile",
     "watch": "gulp watch",
-    "lint": "gulp lint",
+    "lint": "eslint --quiet --cache",
```
Comment (Benjin) [inline package.json:56]: does `--fix` work with this, too?

File: src/connectionconfig/connectionDialogWebviewController.ts [hunk 2]
```diff
@@ -26,7 +26,10 @@ import {
 import { ConnectionOption } from "azdata";
 import { Logger } from "../models/logger";
 import VscodeWrapper from "../controllers/vscodeWrapper";
-import { ConnectionDialog as Loc, refreshTokenLabel } from "../constants/locConstants";
+import {
```
Comment (Benjin) [inline src/connectionconfig/connectionDialogWebviewController.ts:29]: Are there any code changes in these files, or is it just formatting?

**Review by Benjin — APPROVED** [review]

**Review by Benjin — COMMENTED** [review]
ADO PR build is failing due to looking for the lint task; make sure to update that, too.

File: package.json (hunk 1 above)
Comment (aasimkhan30) [reply package.json:56]: Yeah, fix should work as before.

File: src/connectionconfig/connectionDialogWebviewController.ts (hunk 2 above)
Comment (aasimkhan30) [reply src/connectionconfig/connectionDialogWebviewController.ts:29]: Just prettier formatting.

**Pull request merged**